### ISSUE TITLE
RuntimeError: Commit acbd1234 doesn't exist. Are you running `danger local/pr` against the correct repository?

### ISSUE THREAD

**Issue by Vacxe** [danger/danger#1329]
# Report

## What did you do?

```
#!/bin/bash
branch="%teamcity.build.branch%"
export DANGER_GITHUB_HOST=%ghe.hostname%
export DANGER_GITHUB_API_BASE_URL=https://${DANGER_GITHUB_HOST}/api/v3
export GITHUB_REPO_SLUG=%ghe.repo.slug%
export GITHUB_REPO_URL=https://${DANGER_GITHUB_HOST}/${GITHUB_REPO_SLUG}.git
 
export GITHUB_PULL_REQUEST_ID=${branch#"pull/"}


echo "Danger exec"
bundle install
bundle exec danger || true
```

## What did you expect to happen?

And this happens only if current pull request on Head of master branch. 

```
  Results:
09:44:15    [32m
09:44:15   Messages: [0m
09:44:15   - [ ] Ticket: [ABC-0000](https://jira/browse/ABC-0000
09:44:16   Process exited with code 0
```

## What happened instead?

However if any another PR has been merged I receiving message above, where `acbd1234` is latest merged commit 
```

13:41:59   Host key verification failed.
13:41:59   fatal: Could not read from remote repository.
13:41:59   
13:41:59   Please make sure you have the correct access rights
13:41:59   and the repository exists.
13:41:59   Host key verification failed.
13:41:59   fatal: Could not read from remote repository.
13:41:59   
13:41:59   Please make sure you have the correct access rights
13:41:59   and the repository exists.
13:41:59   bundler: failed to load command: danger (/usr/local/bundle/bin/danger)
13:41:59   RuntimeError: Commit acbd1234 doesn't exist. Are you running `danger local/pr` against the correct repository? Also this usually happens when you rebase/reset and force-pushed.
13:41:59     /usr/local/bundle/gems/danger-8.4.0/lib/danger/scm_source/git_repo.rb:117:in `raise_if_we_cannot_find_the_commit'
13:41:59     /usr/local/bundle/gems/danger-8.4.0/lib/danger/scm_source/git_repo.rb:99:in `ensure_commitish_exists_on_branch!'
13:41:59     /usr/local/bundle/gems/danger-8.4.0/lib/danger/request_sources/github/github.rb:111:in `setup_danger_branches'
13:41:59     /usr/local/bundle/gems/danger-8.4.0/lib/danger/danger_core/environment_manager.rb:58:in `ensure_danger_branches_are_setup'
13:41:59     /usr/local/bundle/gems/danger-8.4.0/lib/danger/danger_core/dangerfile.rb:273:in `setup_for_running'
13:41:59     /usr/local/bundle/gems/danger-8.4.0/lib/danger/danger_core/dangerfile.rb:284:in `run'
13:41:59     /usr/local/bundle/gems/danger-8.4.0/lib/danger/danger_core/executor.rb:29:in `run'
13:41:59     /usr/local/bundle/gems/danger-8.4.0/lib/danger/commands/runner.rb:73:in `run'
13:41:59     /usr/local/bundle/gems/claide-1.0.3/lib/claide/command.rb:334:in `run'
13:41:59     /usr/local/bundle/gems/danger-8.4.0/bin/danger:5:in `<top (required)>'
13:41:59     /usr/local/bundle/bin/danger:23:in `load'
13:41:59     /usr/local/bundle/bin/danger:23:in `<top (required)>'
```

## Your Environment

* Running on Teamcity
* Branch reference `refs/pull/1234/head`
* GITHUB_PULL_REQUEST_ID variable is 1234
---

## In conclusion
* If pull request on head of master branch all looks as expected
* If any other changes happened with master - danger failing with message above

**Comment by Vacxe** [comment]
Solved by copied ssh keys into docker container
